darsh175223/try course button

darsh175223 opened the pull request:
I created a new file in the components folder for the Try Course button. The colors and shapes of the button are in line with the figma design

DM524oigjroigre committed:
Created a 'TRY COURSE' component

DM524oigjroigre committed:
Created a 'TRY COURSE' component

stanleylew5 commented:
make sure you link your issue, also sync your github emails or use one account (whichever one was invited) for your issues

stanleylew5 commented on src/components/try-course-button.tsx:
```diff
@@ -0,0 +1,18 @@
+      {/* Larger shape beneath */}
+      <div className="absolute inset-0 rounded-full bg-[#cff2aa]"></div>
+
+      {/* Main button */}
```
```suggestion

```

stanleylew5 commented on src/components/try-course-button.tsx:
```diff
@@ -0,0 +1,18 @@
+const TryCourseButton = () => {
+  return (
+    <div className="relative inline-block w-[20%] p-1.5">
+      {/* Larger shape beneath */}
```
```suggestion

```

stanleylew5 commented on src/components/try-course-button.tsx:
```diff
@@ -0,0 +1,18 @@
+
+      {/* Main button */}
+      <div className="flex justify-center">
+        <button className="relative w-[96%] rounded-full bg-[#eaf7dc] px-12 py-3 text-xl font-bold uppercase text-black shadow-md transition-all duration-300 ease-in-out hover:shadow-lg">
```
```suggestion
        <button className="relative w-[96%] rounded-full bg-[#eaf7dc] px-12 py-3 text-xl font-bold text-black shadow-md">
```

stanleylew5 replied:
also, use dark mode colors since we're going to start with dark mode first. instead of hard coding #eaf7dc, add the color within tailwind config file. pull from dev first though `git pull origin`

stanleylew5 commented on src/components/try-course-button.tsx:
```diff
@@ -0,0 +1,18 @@
+      {/* Main button */}
+      <div className="flex justify-center">
+        <button className="relative w-[96%] rounded-full bg-[#eaf7dc] px-12 py-3 text-xl font-bold uppercase text-black shadow-md transition-all duration-300 ease-in-out hover:shadow-lg">
+          <span className="relative z-10">TRY COURSE</span>
```
```suggestion
          <p>TRY COURSE</p>
```

stanleylew5 commented on src/components/try-course-button.tsx:
```diff
@@ -0,0 +1,18 @@
+      <div className="flex justify-center">
+        <button className="relative w-[96%] rounded-full bg-[#eaf7dc] px-12 py-3 text-xl font-bold uppercase text-black shadow-md transition-all duration-300 ease-in-out hover:shadow-lg">
+          <span className="relative z-10">TRY COURSE</span>
+          <div className="absolute inset-0 rounded-full bg-gradient-to-b from-transparent to-[#eaf7dc] opacity-50"></div>
```
```suggestion
        
```
not quite sure what this is for it does nothing

stanleylew5 commented on src/components/try-course-button.tsx:
```diff
@@ -0,0 +1,18 @@
+const TryCourseButton = () => {
+  return (
+    <div className="relative inline-block w-[20%] p-1.5">
```
```suggestion
    <div className="relative w-[20%] p-1.5">
```

stanleylew5 reviewed (changes requested):
tldr, this code overcomplicates the issue it can be something similar to the below.obviously some styling needs to be done but it takes care of all the absolute inset positioning and all that you're going through right now. try following the code above this comment. and import your colors within the tailwind config file 

```
<div className="bg-green-500 rounded-full py-2 px-4">
        <button className="text-2xl bg-[#eaf7dc] rounded-full px-4 py-2 text-black">TRY COURSE</button>
</div>
```

darsh175223 committed:
Merge branch 'dev' of https://github.com/acm-ucr/developer-station into darsh175223/try-course-button

darsh175223 committed:
testing github config

darsh175223 committed:
Update src/components/try-course-button.tsx

remove comment

Co-authored-by: Stanley Lew <135681592+stanleylew5@users.noreply.github.com>

darsh175223 committed:
Update src/components/try-course-button.tsx

delete comment

Co-authored-by: Stanley Lew <135681592+stanleylew5@users.noreply.github.com>

darsh175223 committed:
Update src/components/try-course-button.tsx

remove the transition

Co-authored-by: Stanley Lew <135681592+stanleylew5@users.noreply.github.com>

darsh175223 committed:
Update src/components/try-course-button.tsx

changing the tag to <p>

Co-authored-by: Stanley Lew <135681592+stanleylew5@users.noreply.github.com>

darsh175223 committed:
Update src/components/try-course-button.tsx

removed inline-block

Co-authored-by: Stanley Lew <135681592+stanleylew5@users.noreply.github.com>

darsh175223 committed:
Update src/components/try-course-button.tsx

removed div

Co-authored-by: Stanley Lew <135681592+stanleylew5@users.noreply.github.com>

darsh175223 committed:
applied new changes to styling

darsh175223 committed:
completed merge conflict

darsh175223 committed:
Changed the colors to dark mode

darsh175223 committed:
Changed the colors to dark mode

darsh175223 committed:
Changed the colors to dark mode

darsh175223 committed:
Changed the colors to dark mode

darsh175223 committed:
Merge branch 'dev' into darsh175223/try-course-button

darsh175223 commented:
I changed the colors to dark mode, removed all the tags that don't serve a purpose and changed the text and the widths to be dependent on the screen size(ex: text-[5vw]).

stanleylew5 committed:
size switch
